### ISSUE TITLE
Enhancement: [@typescript-eslint/parameter-properties] should detect unnecessary readonly property initializers

### ISSUE THREAD

**Issue by Pyrolistical** [typescript-eslint/typescript-eslint#7744]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://main--typescript-eslint.netlify.app/rules/parameter-properties#prefer

### Description

With the following rule configuration:
```json
"@typescript-eslint/parameter-properties": [
  "error",
  { "prefer": "parameter-property" }
]
```
Properties marked as `readonly` in class constructors can omit trivial initializers (`this.x = x`). When the rule is configured as such, it should error and tell users to delete trivial initializers. This is auto-fixable.

### Fail

```typescript
class Foo {
  constructor(readonly a: string, readonly b: string) {
    this.a = a  // only trivial initializers should error
    this.b = String(b.length)
  }
}
```

### Pass

```typescript
class Foo {
  constructor(readonly a: string, readonly b: string) {
    this.b = String(b.length)
  }
}
```

### Additional Info

TypeScript reference that shows readonly trivial initializers are unnecessary https://www.typescriptlang.org/docs/handbook/2/classes.html#parameter-properties

Property are required to be initialized in the constructor if `readonly` is applied to the class property.

**Comment by JoshuaKGoldberg** [comment]
`@typescript-eslint/parameter-properties`just enforces using one style or the other. Detecting unnecessary initializers is a separate concern from the rule.

It looks like #7045 already exists as a feature request for the unnecessary property initialization detection. Marking this issue as a duplicate, but please do post back or file a new one if that's not right. Thanks for filing!
